jumpcloud-ADMU/Powershell/ProgressForm.ps1

```diff
@@ -0,0 +1,193 @@
+                </ScrollViewer>
+            </Expander.Content>
+        </Expander>
+        <Image Name="JCLogoImg" Margin="102,41,102,362" Stretch="Fill"/>
+    </Grid>
+</Window>
+"@
+
+    # Create a runspace to run the form in
+    $newRunspace.ApartmentState = "STA"
+    $newRunspace.ThreadOptions = "ReuseThread"
+    $data = $newRunspace.Open() | Out-Null
+    $newRunspace.SessionStateProxy.SetVariable("syncHash", $syncHash)
+
+    # Add the form code to the powershell instance
+    $psCommand = [PowerShell]::Create().AddScript({
+        # Load an xaml form
+
+        $syncHash.Window = [Windows.Markup.XamlReader]::parse( $SyncHash.XAML )
+        ([xml]$SyncHash.XAML).SelectNodes("//*[@Name]") | % { $SyncHash."$($_.Name)" = $SyncHash.Window.FindName($_.Name) }
+        # Image
+        $SyncHash.JCLogoImg.Source = $syncHash.base64JCLogo
+
+        # TODO: Icon
+        #$SyncHash.Window.Icon = $syncHash.base64JCLogo
+        # Automatically scroll to the bottom of the log
+
+        # Scroll to end of Log
+        $syncHash.ScrollLog.ScrollToEnd()
+
+        $updateForm = {
+            $SyncHash.Window.Title = "JumpCloud ADMU 2.7.0"
+            if ($SyncHash.closeWindow -eq $True) {
+                $syncHash.Window.Close()
+                [System.Windows.Forms.Application]::Exit()
+                Break
+            }
+            # IF close window button is clicked
+            if ($SyncHash.Closing) {
+                $SyncHash.Window.Close()
+                [System.Windows.Forms.Application]::Exit()
+                # Break out of the loop
+                Break
+                # Run Start-JCADMU
+            }
+            if ($SyncHash.logLevel -eq "Error") {
+                $syncHash.Status.Foreground = "Red"
+            }
+            if ($synchash.PercentComplete -eq 100) {
+                $SyncHash.ViewLog.Visibility = "Visible"
+                $SyncHash.StartJCADMU.Visibility = "Visible"
+            }
+
+            # Update Log TextBlock
+            $SyncHash.LogTextBlock.Text = $syncHash.LogText
+            # Update Progress Bar
+            $SyncHash.ProgressBar.Value = $SyncHash.PercentComplete
+
+            # Update Status Text
+            $SyncHash.Status.Text = $SyncHash.StatusInput
+            if ($SyncHash.SecondsRemainingInput) {
+                $TimeRemaining = [System.TimeSpan]::FromSeconds($SyncHash.SecondsRemainingInput)
+                $SyncHash.TimeRemaining.Text = '{0:00}:{1:00}:{2:00}' -f $TimeRemaining.Hours, $TimeRemaining.Minutes, $TimeRemaining.Seconds
+            }
+        }
+        # View Log Button
+        $SyncHash.ViewLog.Add_Click({
+            # Open log \Windows\Temp\jcAdmu.log
+            $scriptPath = "$(Get-WindowsDrive)\Windows\Temp\jcAdmu.log"
+            # Open the log
+           Invoke-Item -Path:($scriptPath)
+       })
+        # Start JCADMU Button
+        $syncHash.StartJCADMU.Add_Click({
+            Set-ExecutionPolicy Bypass -Scope Process -Force
```

Comment:
Set-ExecutionPolicy? Why is this needed? I'd like to avoid changing this within a script

Reply:
Changed execution by file path to restart process